.github/workflows/installation.yaml

```diff
@@ -35,4 +35,5 @@ jobs:
 
       - name: Install dependencies
         run: |
-          uv pip install --system -e .
```

Comment:
Either this or you can also say `activate-environment: true` in `with` block of the install uv step; I wanted to do this later after you merge this PR, to create a common github action file and put the whole installation logic with `activate-environment: true`   there which means we don't need to prefix every command with `uv`